### PR TITLE
TRUNK-5332: HibernateProviderDAO.getCountOfProviders handles more than one match.

### PR DESCRIPTION
**Pull request by samuelmale**
Ticket: https://issues.openmrs.org/browse/TRUNK-5332

### CODE REVIEW

File: api/src/test/java/org/openmrs/api/ProviderServiceTest.java [hunk 1]
```diff
@@ -478,12 +478,32 @@ public void getCountOfProviders_shouldFetchNumberOfProviderMatchingGivenQuery()
...
+	  personName.setPreferred(true);
+	  person.addName(personName);
+	  person = Context.getPersonService().savePerson(person);
+	  Provider provider = new Provider();
+	  provider.setPerson(person);
+	  Context.getProviderService().saveProvider(provider);
+	  
+	  // verif
+	  assertEquals(2, service.getCountOfProviders("Hippo").intValue());
+	}
```
Comment (mks-d) [inline api/src/test/java/org/openmrs/api/ProviderServiceTest.java]: @samuelmale, this test was only added to demonstrate the bug. Now that you have re-enabled the two tests that were `@Ignore`'d, this one can go away.

File: api/src/main/java/org/openmrs/api/db/hibernate/HibernateProviderDAO.java [hunk 2]
```diff
@@ -241,9 +241,8 @@ private Junction getNameSearchExpression(String name) {
 	 */
 	@Override
 	public Long getCountOfProviders(String name, boolean includeRetired) {
-		Criteria criteria = prepareProviderCriteria(name, includeRetired);
-		criteria.setProjection(Projections.countDistinct("providerId"));
-		return (Long) criteria.uniqueResult();
+	  Criteria criteria = prepareProviderCriteria(name, includeRetired);
+	  return (long) criteria.list().size();
```
Comment (mks-d) [inline api/src/main/java/org/openmrs/api/db/hibernate/HibernateProviderDAO.java]: Your formatting is different here, are you sure that you ran `mvn clean package` _before_ making the PR?

Comment (samuelmale) [reply api/src/main/java/org/openmrs/api/db/hibernate/HibernateProviderDAO.java]: I ran a clean install and package before making a PR :)

File: api/src/test/java/org/openmrs/api/ProviderServiceTest.java (hunk 1 above)
Comment (samuelmale) [reply api/src/test/java/org/openmrs/api/ProviderServiceTest.java]: Yes, I was gonna do that before. Let me make more changes :)

File: api/src/main/java/org/openmrs/api/db/hibernate/HibernateProviderDAO.java (hunk 2 above)
Comment (samuelmale) [reply api/src/main/java/org/openmrs/api/db/hibernate/HibernateProviderDAO.java]: sorry, I only ran a "mvn install" sorry.

Comment (samuelmale) [reply api/src/main/java/org/openmrs/api/db/hibernate/HibernateProviderDAO.java]: I have just realised it :(

**Comment by samuelmale** [comment]
@mks-d do you have any idea of why this has failed(terminated) https://travis-ci.org/openmrs/openmrs-core/builds/338941999?utm_source=github_status&utm_medium=notification

**Comment by samuelmale** [comment]
@mks-d  I think this is good now :)

File: api/src/test/java/org/openmrs/api/ProviderServiceTest.java [hunk 3]
```diff
@@ -492,9 +491,8 @@ public void getCountOfProviders_shouldExcludeRetiredProviders() {
 	 * @see ProviderService#getCountOfProviders(String,null)
 	 */
 	@Test
-	@Ignore
 	public void getCountOfProviders_shouldIncludeRetiredProvidersIfIncludeRetiredIsSetToTrue() {
-		assertEquals(4, service.getCountOfProviders("provider").intValue());
+		assertEquals(4, service.getCountOfProviders("provider",true).intValue());
```
Comment (mks-d) [inline api/src/test/java/org/openmrs/api/ProviderServiceTest.java]: This is quite nit picky but could you leave a space after the comma and before `true`?

See **8.2 Blank Spaces** [here](http://www.oracle.com/technetwork/java/javase/documentation/codeconventions-141388.html) Oracle's 'Code Conventions for the Java Programming Language':
>A blank space should appear after commas in argument lists.

Comment (samuelmale) [reply api/src/test/java/org/openmrs/api/ProviderServiceTest.java]: @mks-d  done as expected :)

**Comment by samuelmale** [comment]
Am so sorry @mks-d , this was really a lazy mistake. I think I have updated it :)

**Pull request merged**